net/redisclient.go

```diff
@@ -87,6 +101,96 @@ const (
 	defaultConnMetricsInterval = 60 * time.Second
 )
 
+// JumpHash
+// https://arxiv.org/pdf/1406.2294.pdf
+type JumpHash struct {
+	hash   hash.Hash64
+	shards []string
+}
+
+func NewJumpHash(shards []string) redis.ConsistentHash {
+	return &JumpHash{
+		hash:   fnv.New64(),
+		shards: shards,
+	}
+}
+
+func (j *JumpHash) Get(k string) string {
+	_, err := j.hash.Write([]byte(k))
+	if err != nil {
+		log.Fatalf("Failed to write %s to hash: %v", k, err)
+	}
+
+	key := j.hash.Sum64()
+	h := jump.Hash(key, len(j.shards))
+	return j.shards[int(h)%len(j.shards)]
+}
+
+// Multi-probe consistent hashing - mpchash
+// https://arxiv.org/pdf/1505.00062.pdf
+type multiprobe struct {
+	hash   *mpchash.Multi
+	shards []string
+}
+
+func NewMultiprobe(shards []string) redis.ConsistentHash {
+	return &multiprobe{
+		// 2 seeds and k=21
+		hash: mpchash.New(shards, siphash64seed, [2]uint64{1, 2}, 21),
+		// 2 seeds and k=41
+		//hash:   mpchash.New(shards, siphash64seed, [2]uint64{1, 2}, 41),
```

Comment:
Do we need this commented out block?

Reply:
no it was just for testing, did not help to increase k.

Reply:
cleaned up and made a comment